openfe/tests/conftest.py

```diff
@@ -12,30 +12,62 @@
 from gufe import SmallMoleculeComponent, LigandAtomMapping
 
 
+class SlowTest:
+    """Plugin for a fixture that skips slow tests"""
+
+    def __init__(self, config):
+        self.config = config
+
+    def pytest_collection_modifyitems(self, items, config):
+        if (config.getoption('--runslow') or
+            os.getenv("OFE_SLOW_TESTS", default="false").lower() == 'true'):
+            return
+
+        msg = ("need --runslow pytest cli option or the environment variable "
+           "`OFE_SLOW_TESTS` set to `True` to run")
+        skip_slow = pytest.mark.skip(reason=msg)
+        for item in items:
+            if "slow" in item.keywords:
+                item.add_marker(skip_slow)
+
+
+class IntegrationTest:
+    """Plugin for a fixture that skips very slow integration tests"""
+
+    def __init__(self, config):
+        self.config = config
+
+    def pytest_collection_modifyitems(self, items, config):
+        if (config.getoption('--integration') or
+            os.getenv("OFE_INTEGRATION_TESTS", default="false").lower() == 'true'):
+            return
+
+        msg = ("need --integration pytest cli option or the environment "
+               "variable `OFE_INTEGRATION_TESTS` set to `True` to run")
+        skip_int = pytest.mark.skip(reason=msg)
+        for item in items:
+            if "integration" in item.keywords:
+                item.add_marker(skip_int)
+
+
 # allow for optional slow tests
 # See: https://docs.pytest.org/en/latest/example/simple.html
 def pytest_addoption(parser):
     parser.addoption(
         "--runslow", action="store_true", default=False, help="run slow tests"
     )
+    parser.addoption(
+        "--integration", action="store_true", default=False,
+        help="run long integration tests",
+    )
 
 
 def pytest_configure(config):
-    config.addinivalue_line("markers", "slow: mark test as slow to run")
-
-
-def pytest_collection_modifyitems(config, items):
-    if (config.getoption("--runslow") or
-        os.getenv("OFE_SLOW_TESTS", default="false").lower() == 'true'):
-        # --runslow given in cli or OFE_SLOW_TESTS set to True in env vars
-        # do not skip slow tests
-        return
-    msg = ("need --runslow pytest cli option or the environment variable "
-           "`OFE_SLOW_TESTS` set to `True` to run")
-    skip_slow = pytest.mark.skip(reason=msg)
-    for item in items:
-        if "slow" in item.keywords:
-            item.add_marker(skip_slow)
+    config.pluginmanager.register(SlowTest(config), "slow")
+    config.addinivalue_line("markers", "slow: mark test as slow")
+    config.pluginmanager.register(IntegrationTest(config), "integration")
+    config.addinivalue_line(
+            "markers", "integration: mark test as long integration test")
```

Comment:
shouldn't `--integration` also run the slows tests?

Reply:
Fixed